File: src/main/java/intro/to/design/patterns/using/javalang/factorymethod/idcard/IDCardFactory.java
package intro.to.design.patterns.using.javalang.factorymethod.idcard;

import intro.to.design.patterns.using.javalang.factorymethod.framework.Factory;
import intro.to.design.patterns.using.javalang.factorymethod.framework.Product;

public class IDCardFactory extends Factory {
    @Override
    protected Product createProduct(String owner) {
        return new IDCard(owner);
    }

    @Override
    protected void registerProduct(Product product) {
        System.out.println("now registered " + product);
    }
}

File: typescript-design-pattern/src/into-to-design-patterns-using-java-lang/factory-method/id-card/id-card-factory.ts
import Output from '../../../utils/stdin-stdout/output'
import Factory from '../framework/factory'
import { Product } from '../framework/product'
import IdCard from '.'

export default class IdCardFactory extends Factory {
  createProduct(owner: string): Product {
    return new IdCard(owner)
  }

  registerProduct(product: Product): void {
    Output.dir({ '->': 'now registered ', product })
  }
}
